Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,206 @@
+import UIKit
+@_implementationOnly import MapboxCommon_Private
+@_implementationOnly import MapboxCoreMaps_Private
+
+public protocol ViewAnnotationManager: AnyObject {
+    // TODO: Add documentation
+    func addViewAnnotation(_ view: UIView, _ options: ViewAnnotationOptions) throws -> AnnotationView
+    func removeViewAnnotation(_ annotatonView: AnnotationView) -> Bool
+    func updateViewAnnotation(_ annotatonView: AnnotationView, _ options: ViewAnnotationOptions) -> Bool
+    func getViewAnnotation(byFeatureId identifier: String) -> AnnotationView?
+    func getViewAnnotationOptions(byFeatureId identifier: String) -> ViewAnnotationOptions?
+    func getViewAnnotationOptions(byAnnotationView view: AnnotationView) -> ViewAnnotationOptions?
+}
+
+public enum ViewAnnotationManagerError: Error {
+    case geometryFieldMissing
+}
+
+internal class ViewAnnotationManagerImpl: ViewAnnotationManager {
+
+    private let mapboxMap: MapboxMapProtocol
+    private let viewAnnotationWrapper = UIView(frame: .zero)
+    private var annotationViewsById: [String: AnnotationView] = [:]
+    private var annotationIdByViews: [AnnotationView: String] = [:]
+
+    internal init(view: UIView, mapboxMap: MapboxMapProtocol) {
+        view.insertSubview(viewAnnotationWrapper, at: 1)
+        viewAnnotationWrapper.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            viewAnnotationWrapper.topAnchor.constraint(equalTo: view.topAnchor),
+            viewAnnotationWrapper.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            viewAnnotationWrapper.leftAnchor.constraint(equalTo: view.leftAnchor),
+            viewAnnotationWrapper.rightAnchor.constraint(equalTo: view.rightAnchor)
+        ])
+
+        self.mapboxMap = mapboxMap
+        mapboxMap.setViewAnnotationPositionsUpdateListenerFor(listener: self)
+    }
+
+    // MARK: - ViewAnnotationManagerInterface protocol functions
+
+    public func addViewAnnotation(_ view: UIView, _ options: ViewAnnotationOptions) throws -> AnnotationView {
```

Comment:
Could we flip the terminology to `annotationView`? I think it would make all these methods read more fluidly.

Reply:
```suggestion
    public func addAnnotationView(withContentView view: UIView, _ options: ViewAnnotationOptions) throws -> AnnotationView {
```

Reply:
Maybe something like that